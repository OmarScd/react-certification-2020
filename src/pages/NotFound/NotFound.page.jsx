import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import Layout from '../../components/Layout';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
`;

const Message = styled.p`
  font-size: ${(props) => props.theme.fontSizes.medium};
`;

const HomeLink = styled(Link)`
  color: ${(props) => props.theme.colors.font};
  font-size: ${(props) => props.theme.fontSizes.small};
  font-weight: bold;
  transition: color 0.3s ease-in-out;
`;

function NotFoundPage() {
  return (
    <Layout>
      <Container>
        <div>
          <img src="/404.gif" alt="page not found" />
        </div>
        <Message>This is not the page you are lookin for</Message>
        <HomeLink to="/">Return to home page</HomeLink>
      </Container>
    </Layout>
  );
}

export default NotFoundPage;
