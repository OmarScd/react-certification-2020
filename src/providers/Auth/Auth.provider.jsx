import React, { useState, useEffect, useContext } from 'react';

import { USERNAME_STORAGE_KEY } from '../../utils/constants';
import { storage } from '../../utils/storage';

const AuthContext = React.createContext(null);

const useAuth = () => {
  const context = useContext(AuthContext);
  if (!context) {
    throw new Error(`Can't use "useAuth" without an AuthProvider!`);
  }
  return context;
};

const AuthProvider = ({ children }) => {
  const [authenticated, setAuthenticated] = useState(false);
  const [username, setUsername] = useState(undefined);

  useEffect(() => {
    const user = storage.get(USERNAME_STORAGE_KEY);
    const isAuthenticated = !!user;

    setAuthenticated(isAuthenticated);
    if (isAuthenticated) {
      setUsername(user);
    }
  }, []);

  const login = (name) => {
    setAuthenticated(true);
    setUsername(name);
    storage.set(USERNAME_STORAGE_KEY, name);
  };

  const logout = () => {
    setAuthenticated(false);
    setUsername(undefined);
    storage.remove(USERNAME_STORAGE_KEY);
  };

  return (
    <AuthContext.Provider value={{ login, logout, authenticated, username }}>
      {children}
    </AuthContext.Provider>
  );
};

export { useAuth, AuthProvider as default };
