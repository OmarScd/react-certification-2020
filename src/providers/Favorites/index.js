export { default } from './Favorites.provider';
