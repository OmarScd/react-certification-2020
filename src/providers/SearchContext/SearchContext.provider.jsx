import React, { useState, useContext } from 'react';

const SearchContext = React.createContext();

const useSearch = () => {
  const context = useContext(SearchContext);
  if (!context) {
    throw new Error(`Can't use "useSearch" without a FavoritesProvider!`);
  }
  return context;
};

const SearchContextProvider = ({ children }) => {
  const [query, setQuery] = useState('Subdivisions');

  return (
    <SearchContext.Provider value={{ query, setQuery }}>
      {children}
    </SearchContext.Provider>
  );
};

export { useSearch, SearchContextProvider as default };
