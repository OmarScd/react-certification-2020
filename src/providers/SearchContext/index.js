export { default, useSearch } from './SearchContext.provider';
