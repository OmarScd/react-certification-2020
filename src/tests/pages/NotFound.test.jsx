import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NotFound from '../../pages/NotFound';
import ThemeProvider from '../../providers/Theme';
import '../../providers/SearchContext';
import '../../providers/Auth';

jest.mock('../../providers/SearchContext', () => ({
  default: jest.fn(),
  useSearch: () => ({ query: 'test', setQuery: jest.fn() }),
}));

jest.mock('../../providers/Auth', () => ({
  default: jest.fn(),
  useAuth: () => ({ authenticated: true }),
}));

describe('NotFound page', () => {
  afterAll(cleanup);

  it('renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <BrowserRouter>
          <NotFound />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
