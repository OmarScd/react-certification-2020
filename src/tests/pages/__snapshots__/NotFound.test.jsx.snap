// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page renders correctly 1`] = `
<div
  class="container"
>
  <header
    class="sc-hKFyIo jcdTAl"
  >
    <div
      class="sc-eCApGN hCdAiT"
    >
      <div
        class="sc-jSFkmK qmXvl"
      >
        <a
          class="sc-gKAblj iUhdyg"
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="sc-jSFkmK qmXvl"
      >
        <a
          class="sc-gKAblj iUhdyg"
          href="/favorites"
        >
          Favorites
        </a>
      </div>
      <div
        class="sc-jSFkmK qmXvl"
      >
        <form
          class="sc-bdnylx hYwFxl"
        >
          <img
            alt="magnifying lens"
            class="sc-dlnjPT dGgzTM"
            src="search-24px.svg"
          />
          <input
            class="sc-gtssRu fkigkU"
            placeholder="Search..."
            type="text"
            value="test"
          />
        </form>
      </div>
    </div>
    <div
      class="sc-eCApGN hCdAiT"
    >
      <div
        class="sc-jSFkmK qmXvl"
      >
        <button
          class="sc-iCoHVE kHdmrf"
        >
          Log out
        </button>
      </div>
      <div
        class="sc-jSFkmK qmXvl"
      >
        <img
          alt="person"
          class="sc-fujyUd hfIOzZ"
          src="person-24px.svg"
        />
      </div>
    </div>
  </header>
  <div
    class="content"
  >
    <div
      class="sc-pNWxx eWBObX"
    >
      <div>
        <img
          alt="page not found"
          src="404.gif"
        />
      </div>
      <p
        class="sc-jrsJCI sdrJr"
      >
        This is not the page you are lookin for
      </p>
      <a
        class="sc-kEqYlL guMqYO"
        href="/"
      >
        Return to home page
      </a>
    </div>
  </div>
</div>
`;
